### DISCUSSION TITLE
should `RegexBuilder::case_sensitive(false)` be added as an alias for `RegexBuilder::case_insensitive(true)`?

### DISCUSSION THREAD

**Discussion by dciug** [rust-lang/regex#1209] [Q&A] [answered]
Would it make sense to perhaps add a separate function to configure the case sensitivity of a regex?
To my mind, it's easier to read `RegexBuilder::new("").case_sensitive(true)` instead of `RegexBuilder::new("").case_insensitive(false)`.

Related blog: https://testing.googleblog.com/2023/10/improve-readability-with-positive.html
Related method: https://docs.rs/regex/latest/regex/struct.RegexBuilder.html#method.case_insensitive

**Comment by BurntSushi** [comment, marked as answer]
If I were starting over, I might be inclined to use `case_sensitive` rather than `case_insensitive`. However, crucially, the actual regex flag is `i`, which means "opt into case insensitive matching." In this way, `RegexBuilder` is matching the conceptualization prompted by the syntax. And changing the `i` to something else is a much more difficult change. For example, since case sensitive matching is generally expected to be enabled by default, it would imply doing something like `(?-X)` (where `X` is the flag that means "enable case sensitive matching") to get case insensitive matching.

But as for now, no, I actually think having both is worse than having one. When you have both, folks start wondering what the difference is between them and which one they should use.
